options/rtdl/generic/main.cpp

```diff
@@ -179,10 +187,15 @@ extern "C" void *interpreterMain(uintptr_t *entry_stack) {
 	aux += *aux + 1; // First, we skip argc and all args.
 	__ensure(!*aux);
 	aux++;
-	while(*aux) // Now, we skip the environment.
-		aux++;
+
+	rtdl_envp = reinterpret_cast<char **>(aux);
```

Comment:
The values for these configuration options should simply be saved early, rather than requeried each time. This would also prevent having to think about whether envp is initialized at all.

Reply:
Agreed. We should only process the environment once.